tests/regression/test_interpolate.py

```diff
@@ -358,3 +358,126 @@ def test_interpolate_periodic_coords_max():
+        super(FIAT_P3IntMoments, self).__init__(poly_set, dual_set, 3, 0)
+
+
+def test_fiat_p3intmoments():
+    # can create
+    el = FIAT_P3IntMoments(UFCInterval(), 3)
+    # dual eval gives expected values from sum of point evaluations
+    fns = (lambda x: x[0], lambda x: x[0]**2)
+    expected = ([0, 1, 1/2, 1/3], [0, 1, 1/3, 1/4])
+    for i in range(len(fns)):
+        node_vals = []
+        for node in el.dual.nodes:
+            pt_dict = node.pt_dict
+            node_val = 0.0
+            for pt in pt_dict:
+                for (w, _) in pt_dict[pt]:
+                    node_val += w * fns[i](pt)
+            node_vals.append(node_val)
+        assert len(node_vals) == len(expected[i])
+        assert np.allclose(node_vals, expected[i])
+
+
+# FInAT Equivalent
+
+from finat.fiat_elements import ScalarFiatElement
+
+
+class FInAT_P3IntMoments(ScalarFiatElement):
+    def __init__(self, cell, degree):
+        super(FInAT_P3IntMoments, self).__init__(FIAT_P3IntMoments(cell, degree))
+
+
+# Replace the old tsfc.finatinterface.convert dispatch with a new one that
+# gives the the new FInAT element for P3 on an interval with variant
+# "interior-moment"
+
+from tsfc.finatinterface import convert, as_fiat_cell
+import ufl
+old_convert = convert.dispatch(ufl.FiniteElement)
+
+
+def temp_convert(element, **kwargs):
+    if element.family() == "Lagrange" and element.cell() == interval and element.degree() == 3 and element.variant() == "interior-moment":
+        return FInAT_P3IntMoments(as_fiat_cell(element.cell()), element.degree()), set()
+    else:
+        return old_convert(element, **kwargs)
+
+
+# Register the new tsfc covert method - remove after tests have run (post yield)
+@pytest.fixture
+def add_p3intmoments_tsfc():
+    convert.register(ufl.FiniteElement, temp_convert)
+    yield
+    convert.register(ufl.FiniteElement, old_convert)
+
+
+# Test New Element Dual Evaluation
+def test_basic_dual_eval_p3intmoments(add_p3intmoments_tsfc):
+    mesh = UnitIntervalMesh(1)
+    e = ufl.FiniteElement("CG", "interval", 3, variant="interior-moment")
+    V = FunctionSpace(mesh, e)
+    x = SpatialCoordinate(mesh)
+    expr = Constant(1.)
+    f = interpolate(expr, V)
+    assert np.allclose(f.dat.data_ro[f.cell_node_map().values], [node(expr) for node in f.function_space().finat_element.fiat_equivalent.dual_basis()])
+    expr = x[0]
+    # Account for cell and corresponding expression being flipped onto
+    # reference cell before reaching FIAT
+    expr_fiat = 1-x[0]
+    f = interpolate(expr, V)
+    assert np.allclose(f.dat.data_ro[f.cell_node_map().values], [node(expr_fiat) for node in f.function_space().finat_element.fiat_equivalent.dual_basis()])
+    expr = x[0]**2
+    expr_fiat = (1-x[0])**2
+    f = interpolate(expr, V)
+    assert np.allclose(f.dat.data_ro[f.cell_node_map().values], [node(expr_fiat) for node in f.function_space().finat_element.fiat_equivalent.dual_basis()])
```

Comment:
Can we put all this new stuff in a separate file, rather than with `test_interpolate.py`?